compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/method-call-computed.expect.md

```diff
@@ -33,26 +33,18 @@ function foo(a, b, c) {
     t0 = $[1];
   }
   const x = t0;
+  const y = makeObject(a);
   let t1;
-  if ($[2] !== a) {
-    t1 = makeObject(a);
-    $[2] = a;
-    $[3] = t1;
-  } else {
-    t1 = $[3];
-  }
-  const y = t1;
-  let t2;
-  if ($[4] !== x || $[5] !== y.method || $[6] !== b) {
-    t2 = x[y.method](b);
-    $[4] = x;
-    $[5] = y.method;
-    $[6] = b;
-    $[7] = t2;
+  if ($[2] !== x || $[3] !== y.method || $[4] !== b) {
+    t1 = x[y.method](b);
+    $[2] = x;
+    $[3] = y.method;
+    $[4] = b;
+    $[5] = t1;
   } else {
-    t2 = $[7];
+    t1 = $[5];
   }
-  const z = t2;
+  const z = t1;
```

Comment:
This is a perfect example of how this PR makes escape analysis more precise. The dependency is on `y.method`, which we only use where a primitive is expected (in `x[y.method]`). We don't have to memoize the `makeObject(a)` call if we're only going to extract a primitive from it.